tests/lean/run/ coinductive_predicates_errors.lean

```diff
@@ -0,0 +1,75 @@
+/-!
+This file contains tests for typical mistakes one might do when using `least_fixpoint`/`greatest_fixpoint`/`partial_fixpoint` machinery, that is:
+ - Try to use `greatest_fixpoint`/`least_fixpoint` to define functions instead of predicates
+ - Mix `partial_fixpoint` with `greatest_fixpoint`/`least_fixpoint` in the same clique
+ - Apply `greatest_fixpoint`/`least_fixpoint` to non-recursive functions
+-/
+
+/--
+error: `greatest_fixpoint` can be only used to define predicates
+-/
+#guard_msgs in
+def f (x : Nat) : Nat :=
+  f (x + 1)
+greatest_fixpoint
+
+/--
+error: `least_fixpoint` can be only used to define predicates
+-/
+#guard_msgs in
+def g (x : Nat) : Nat :=
+  g (x + 1)
+least_fixpoint
+
+/--
+warning: unused `partial_fixpoint`, function is not recursive
+-/
+#guard_msgs in
+def h (x : Nat) : Prop := x > 42
+  greatest_fixpoint
+
+/--
+error: all functions in the clique must be defined using lattice theory, but some are not
```

Comment:
Should we maybe avoid “lattice theory” in user-facing docs?

Also see the test `tests/lean/termination_by.lean` where we have a similar check for not mixing structural and well-founded recursion. It produces a bit more verbose error messages (pointing out which function is defined like what etc.). Did you see that, and did you try to do the same? (The check is performed in `src/Lean/Elab/PreDefinition/Main.lean`)

A test mixing `least_fixpoint` and `termination_by` would also be good :-)

If you feel like it would be useful, maybe move all tests that are about invalid termination hint combinations in mutual groups from here and from `tests/lean/termination_by.lean` into its dedicated `mutual_termination_by_errors.lean` file.